Source/DafnyLanguageServer.Test/Util/ClientBasedLanguageServerTest.cs

```diff
@@ -55,19 +128,28 @@ public class ClientBasedLanguageServerTest : DafnyLanguageServerTestBase, IAsync
     return things.Select(t => t.DocumentSymbol!);
   }
 
-  public async Task<Diagnostic[]> GetLastDiagnostics(TextDocumentItem documentItem, CancellationToken cancellationToken) {
+  public async Task<PublishDiagnosticsParams> GetLastDiagnosticsParams(TextDocumentItem documentItem, CancellationToken cancellationToken) {
     await client.WaitForNotificationCompletionAsync(documentItem.Uri, cancellationToken);
-    var document = (await Projects.GetLastDocumentAsync(documentItem))!;
-    Assert.NotNull(document);
-    Assert.Equal(documentItem.Version, document.Version);
-    Diagnostic[] result;
-    do {
-      result = await diagnosticsReceiver.AwaitNextDiagnosticsAsync(cancellationToken);
-    } while (!document!.AllFileDiagnostics.Select(d => d.ToLspDiagnostic()).SequenceEqual(result));
+    var compilation = (await Projects.GetLastDocumentAsync(documentItem))!;
+    Assert.NotNull(compilation);
+    Assert.True(documentItem.Version <= compilation.Version);
```

Comment:
Why is okay to allow `documentItem.Version < compilation.Version`?

Reply:
Each time _any_ document in a project changes, the version field of the related ProjectManager is incremented, so that grows strictly faster than the version of any single document part of that project.